src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -807,6 +885,11 @@ public virtual bool Disconnect(DisconnectParams disconnectParams)
                 return false;
             }
 
+            if (info.ConnectionDetails.AuthenticationType == "AzureMFA")

```

Comment:
Is this also called when an editor is closed?

And please add a comment explaining why you're doing this